Whiskey/Functions/Assert-WhiskeyTaskSkip.ps1

```diff
@@ -0,0 +1,183 @@
+        foreach( $wildcard in $Properties['ExceptOnBranch'] )
+        {
+            if( $branch -like $wildcard )
+            {
+                $runTask = $false
+                Write-WhiskeyVerbose -Context $Context -Message ('                {0}     -like  {1}' -f $branch, $wildcard)
+                break
+            }
+
+            Write-WhiskeyVerbose -Context $Context -Message     ('                {0}  -notlike  {1}' -f $branch, $wildcard)
+        }
+        if( -not $runTask )
+        {
+            return $true
+        }
+    }
+
+    $modes = @( 'Clean', 'Initialize', 'Build' )
+    $onlyDuring = $Properties['OnlyDuring']
+    $exceptDuring = $Properties['ExceptDuring']
+
+    if ($onlyDuring -and $exceptDuring)
+    {
+        Stop-WhiskeyTask -TaskContext $Context -Message 'Both ''OnlyDuring'' and ''ExceptDuring'' properties are used. These properties are mutually exclusive, i.e. you may only specify one or the other.'
+        return
+    }
+    elseif ($onlyDuring -and ($onlyDuring -notin $modes))
+    {
+        Stop-WhiskeyTask -TaskContext $Context -Message ('Property ''OnlyDuring'' has an invalid value: ''{0}''. Valid values are: ''{1}''.' -f $onlyDuring,($modes -join "', '"))
+        return
+    }
+    elseif ($exceptDuring -and ($exceptDuring -notin $modes))
+    {
+        Stop-WhiskeyTask -TaskContext $Context -Message ('Property ''ExceptDuring'' has an invalid value: ''{0}''. Valid values are: ''{1}''.' -f $exceptDuring,($modes -join "', '"))
+        return
+    }
+
+    if ($onlyDuring -and ($Context.RunMode -ne $onlyDuring))
+    {
+        Write-WhiskeyVerbose -Context $Context -Message ('OnlyDuring.{0} -ne Build.RunMode.{1}' -f $onlyDuring,$Context.RunMode)
+        return $true
+    }
+    elseif ($exceptDuring -and ($Context.RunMode -eq $exceptDuring))
+    {
+        Write-WhiskeyVerbose -Context $Context -Message ('ExceptDuring.{0} -ne Build.RunMode.{1}' -f $exceptDuring,$Context.RunMode)
+        return $true
+    }
+
+    if( $Properties['IfExists'] )
+    {
+        $exists = Test-Path -Path $Properties['IfExists']
+        if( -not $exists )
+        {
+            Write-WhiskeyVerbose -Context $Context -Message ('IfExists  {0}  not exists' -f $Properties['IfExists']) -Verbose
+            return $true
+        }
+        Write-WhiskeyVerbose -Context $Context -Message     ('IfExists  {0}      exists' -f $Properties['IfExists']) -Verbose
+    }
+
+    if( $Properties['UnlessExists'] )
+    {
+        $exists = Test-Path -Path $Properties['UnlessExists']
+        if( $exists )
+        {
+            Write-WhiskeyVerbose -Context $Context -Message ('UnlessExists  {0}      exists' -f $Properties['UnlessExists']) -Verbose
+            return $true
+        }
+        Write-WhiskeyVerbose -Context $Context -Message     ('UnlessExists  {0}  not exists' -f $Properties['UnlessExists']) -Verbose
+    }
+
+    if( $Properties['OnlyIfBuild'] )

```

Comment:
Rename this to `OnlyIfBuildStatusIs`.  Also, add the reverse, `ExceptIfBuildStatusIs`.

Reply:
Should I rename the Context.BuildStatus properties as well? Currently they are:
```
public enum BuildStatus
{
    NotStarted,
    Started,
    Succeeded,
    Failed
}
```
Which would look like: `OnlyIfBuildStatusIs: Succeeded`

Reply:
No.